base/libgit2/tree.jl

```diff
@@ -16,21 +15,85 @@ function treewalk(f::Function, tree::GitTree, payload=Any[], post::Bool = false)
     return cbf_payload
 end
 
+"Returns a file name, as a `String`, of a tree entry."
 function filename(te::GitTreeEntry)
     str = ccall((:git_tree_entry_name, :libgit2), Cstring, (Ptr{Void},), te.ptr)
-    str != C_NULL && return unsafe_string(str)
-    return nothing
+    str == C_NULL && throw(Error.GitError(Error.Tree,
+                                          LibGit2.Error.ENOTFOUND, "not found"))
```

Comment:
should still say what was not found if this same message is used in multiple places